packages/theme_tailor/lib/src/util/extension/library_element_extension.dart

```diff
@@ -0,0 +1,12 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:theme_tailor/src/util/import_finder.dart';
+
+extension LibraryElementExtension on LibraryElement {
+  bool get hasFlutterDiagnosticableImport {
```

Comment:
Wouldn't it be better to just export it in annotation package so it will be always available?

Reply:
Yep, that would be go to solution but currently it doesn't work. There are some errors when running build runner.
This is the same problem that prevents generator from using Flutter classes like Color/TextStyle etc
<details>
  <summary>Details here</summary>
 
  
  ```console
  [INFO] Generating build script...
[INFO] Generating build script completed, took 412ms

[INFO] Initializing inputs
[INFO] Reading cached asset graph...
[INFO] Reading cached asset graph completed, took 94ms

[INFO] Checking for updates since last build...
[INFO] Checking for updates since last build completed, took 652ms

[WARNING] Invalidating asset graph due to build script update!
[INFO] Cleaning up outputs from previous builds....
[INFO] Cleaning up outputs from previous builds. completed, took 7ms

[INFO] Generating build script...
[INFO] Generating build script completed, took 76ms

[WARNING] Invalidated precompiled build script due to missing asset graph.
[INFO] Precompiling build script......
[WARNING] ../../../../../development/flutter/packages/flutter/lib/src/foundation/basic_types.dart:9:1: Error: Not found: 'dart:ui'
export 'dart:ui' show VoidCallback;
^
../../../../../development/flutter/packages/flutter/lib/src/foundation/binding.dart:8:8: Error: Not found: 'dart:ui'
import 'dart:ui' as ui show SingletonFlutterWindow, Brightness, PlatformDispatcher, window;
       ^
../../../../../development/flutter/packages/flutter/lib/src/foundation/debug.dart:5:8: Error: Not found: 'dart:ui'
import 'dart:ui' as ui show Brightness;
       ^
../../../../../development/flutter/packages/flutter/lib/src/foundation/binding.dart:196:3: Error: Type 'ui.SingletonFlutterWindow' not found.
  ui.SingletonFlutterWindow get window => ui.window;
  ^^^^^^^^^^^^^^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/binding.dart:217:3: Error: Type 'ui.PlatformDispatcher' not found.
  ui.PlatformDispatcher get platformDispatcher => ui.PlatformDispatcher.instance;
  ^^^^^^^^^^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:68:20: Error: Type 'VoidCallback' not found.
  void addListener(VoidCallback listener);
                   ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:72:23: Error: Type 'VoidCallback' not found.
  void removeListener(VoidCallback listener);
                      ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:116:21: Error: Type 'VoidCallback' not found.
  static final List<VoidCallback?> _emptyListeners = List<VoidCallback?>.filled(0, null);
                    ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:117:8: Error: Type 'VoidCallback' not found.
  List<VoidCallback?> _listeners = _emptyListeners;
       ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:183:20: Error: Type 'VoidCallback' not found.
  void addListener(VoidCallback listener) {
                   ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:243:23: Error: Type 'VoidCallback' not found.
  void removeListener(VoidCallback listener) {
                      ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:388:20: Error: Type 'VoidCallback' not found.
  void addListener(VoidCallback listener) {
                   ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:395:23: Error: Type 'VoidCallback' not found.
  void removeListener(VoidCallback listener) {
                      ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/debug.dart:127:1: Error: Type 'ui.Brightness' not found.
ui.Brightness? debugBrightnessOverride;
^^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/binding.dart:196:46: Error: Undefined name 'window'.
  ui.SingletonFlutterWindow get window => ui.window;
                                             ^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/binding.dart:217:54: Error: Undefined name 'PlatformDispatcher'.
  ui.PlatformDispatcher get platformDispatcher => ui.PlatformDispatcher.instance;
                                                     ^^^^^^^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/binding.dart:502:46: Error: Undefined name 'Brightness'.
                debugBrightnessOverride = ui.Brightness.light;
                                             ^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/binding.dart:505:46: Error: Undefined name 'Brightness'.
                debugBrightnessOverride = ui.Brightness.dark;
                                             ^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:68:20: Error: 'VoidCallback' isn't a type.
  void addListener(VoidCallback listener);
                   ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:72:23: Error: 'VoidCallback' isn't a type.
  void removeListener(VoidCallback listener);
                      ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:116:21: Error: 'VoidCallback' isn't a type.
  static final List<VoidCallback?> _emptyListeners = List<VoidCallback?>.filled(0, null);
                    ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:116:59: Error: 'VoidCallback' isn't a type.
  static final List<VoidCallback?> _emptyListeners = List<VoidCallback?>.filled(0, null);
                                                          ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:117:8: Error: 'VoidCallback' isn't a type.
  List<VoidCallback?> _listeners = _emptyListeners;
       ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:183:20: Error: 'VoidCallback' isn't a type.
  void addListener(VoidCallback listener) {
                   ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:187:27: Error: 'VoidCallback' isn't a type.
        _listeners = List<VoidCallback?>.filled(1, null);
                          ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:189:20: Error: 'VoidCallback' isn't a type.
        final List<VoidCallback?> newListeners =
                   ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:190:18: Error: 'VoidCallback' isn't a type.
            List<VoidCallback?>.filled(_listeners.length * 2, null);
                 ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:208:18: Error: 'VoidCallback' isn't a type.
      final List<VoidCallback?> newListeners = List<VoidCallback?>.filled(_count, null);
                 ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:208:53: Error: 'VoidCallback' isn't a type.
      final List<VoidCallback?> newListeners = List<VoidCallback?>.filled(_count, null);
                                                    ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:243:23: Error: 'VoidCallback' isn't a type.
  void removeListener(VoidCallback listener) {
                      ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:250:13: Error: 'VoidCallback' isn't a type.
      final VoidCallback? listenerAtIndex = _listeners[i];
            ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:350:20: Error: 'VoidCallback' isn't a type.
        final List<VoidCallback?> newListeners = List<VoidCallback?>.filled(newLength, null);
                   ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:350:55: Error: 'VoidCallback' isn't a type.
        final List<VoidCallback?> newListeners = List<VoidCallback?>.filled(newLength, null);
                                                      ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:354:17: Error: 'VoidCallback' isn't a type.
          final VoidCallback? listener = _listeners[i];
                ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:388:20: Error: 'VoidCallback' isn't a type.
  void addListener(VoidCallback listener) {
                   ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/change_notifier.dart:395:23: Error: 'VoidCallback' isn't a type.
  void removeListener(VoidCallback listener) {
                      ^^^^^^^^^^^^
../../../../../development/flutter/packages/flutter/lib/src/foundation/debug.dart:127:4: Error: 'Brightness' isn't a type.
ui.Brightness? debugBrightnessOverride;
   ^^^^^^^^^^
[INFO] Precompiling build script... completed, took 1.3s

[SEVERE] Failed to precompile build script .dart_tool/build/entrypoint/build.dart.
This is likely caused by a misconfigured builder definition.

pub finished with exit code 78
  ```
  
</details>

Reply:
Ok, so we will need to live with manual imports.